nodejs/device/src/index.ts

```diff
@@ -175,11 +178,38 @@ class DeviceApp {
         );
       };
       this.twin.on("properties.desired.thief.pairing", listener);
-      const timer = setTimeout(
+      const pairingAttemptTimer = setTimeout(
         on_timeout,
         1000 * settings.pairingRequestSendIntervalInSeconds
       );
-    });
+
+      const finish = () => {
+        logger.warn("Pairing canceled.");
+        this.twin.removeListener("properties.desired.thief.pairing", listener);
+        clearTimeout(pairingAttemptTimer);
+        reject(new Canceled());
+      };
+
+      cancel = () => {
+        if (finished) {
+          return;
+        }
+        finish();
+      };
+
+      if (finished) {
+        finish();
+      }
+    })
+      .then<ThiefPairingProperties>((received) => {
+        finished = true;
+        return received;
+      })
+      .catch<ThiefPairingProperties>((err) => {
+        finished = true;
+        return err;
+      });
+    return { promise, cancel };
```

Comment:
Confusing indent.  I think this needs to be one tabstop to the left of where it is.

Reply:
I'm letting prettier call the shots here.